src/convertToObject.js

```
@@ -1,12 +1,20 @@
 'use strict';
 
-/**
- * @param {string} sourceString
- *
- * @return {object}
- */
 function convertToObject(sourceString) {
-  // write your code here
+  const res = {};
+
+  sourceString
+    .trim()
+    .split(';')
+    .forEach((param) => {
+      const [property, value] = param.split(':');
```

This line splits each parameter by ':', but if the parameter string is empty (for example, due to a trailing semicolon), `param.split(':')` will return `["" ]`, so both `property` and `value` may be undefined. This is not a critical error for this task, but be aware that empty segments are silently ignored by your current logic.